dts/bindings/gpio/gpio-controller.yaml

```diff
@@ -25,3 +25,17 @@ properties:
           "ngpios = <18>;" informs the driver that only the first 18 GPIOs, at
           local offset 0 .. 17, are in use.  For cases in which there might be
           holes in the slot range, this value should be the max slot number-1.
+    gpio-reserved-ranges:
+      type: array
```

Comment:
This property needs additional logic in the DTS scripts to transform the tuple array to bitmask (probably in most drivers). I am wondering if this would need a new type, or how to mark this property to needed the additional conversion translation?

Reply:
@galak Is it your intent to have this bitmask available in the tooling soon enough that I should hold off on working #20115 until the mask is available?  The switch to a mask will require touching every driver.

Reply:
Suggested commit https://github.com/fabled/zephyr/commit/bit-ranges, though I am not yet fully familiar with the DTS parsing code ideology, so not sure if this is the desired location for the conversion. Happy to make PR if it looks good.

Reply:
But seems the `gpio-reserved-ranges` is going to become common code. Maybe I should completely remove that part from my PR #21245 and rebase it on top of #20115 ?